### PR TITLE
chore: integrate session

### PR DESCRIPTION
**Pull request by akaJSaunders**
### Work Checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

---

### Merge Instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,12 +1,19 @@
+import { SessionStore } from 'ch-node-session-handler';
 import * as express from 'express';
+import IORedis from 'ioredis';
 import * as nunjucks from 'nunjucks';
 import * as path from 'path';
 import { routes } from './routes/routes';
 
+import cookieParser = require('cookie-parser');
```
Comment (rhysbarrett) [inline src/app.ts:8]: Should this be `import cookieParser from 'cookie-parser';`?

File: src/app.ts [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+import { ROOT_URI } from './routes/paths';
+import { SessionMiddleware } from './middlewares/SessionMiddleware';
+
 const app = express();
 
 // set up app variables from the environment
 app.set('port', process.env.PORT || '3000');
+app.locals.cdn = { host: process.env.CDN_HOST };
```
Comment (rhysbarrett) [inline src/app.ts:16]: Just as a heads-up, this will conflict with Natalie's changes upon rebase.

File: src/app.ts [hunk 3]
```diff
@@ -1,12 +1,19 @@
+import { SessionStore } from 'ch-node-session-handler';
 import * as express from 'express';
+import IORedis from 'ioredis';
 import * as nunjucks from 'nunjucks';
 import * as path from 'path';
 import { routes } from './routes/routes';
 
+import cookieParser = require('cookie-parser');
+import { ROOT_URI } from './routes/paths';
```
Comment (rhysbarrett) [inline src/app.ts:9]: Is this value being used in this file?

File: src/controllers/StartPageController.ts [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
+import { ROOT_URI } from '../routes/paths';
 
 export class StartPageController {
 
-  public renderView = (req: Request, res: Response, next: NextFunction) => res.render('start-page');
+  public renderView = (req: Request, res: Response) => {
+    res.render('start-page');
+  };
+
+  public route = (req: Request, res: Response) => {
```
Comment (rhysbarrett) [inline src/controllers/StartPageController.ts:10]: What does this function do?

File: tsconfig.json [hunk 5]
```diff
@@ -10,7 +10,7 @@
     "removeComments": true,
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "strictNullChecks": true
+    "allowSyntheticDefaultImports": true
```
Comment (rhysbarrett) [inline tsconfig.json:13]: Have you tried using `esModuleInterop` instead, as per the discussion [here](https://stackoverflow.com/questions/52576203/do-i-ever-need-explicit-allowsyntheticdefaultimports-if-esmoduleinterop-is-true)? This is the parameter used in my branch.

File: tsconfig.json [hunk 6]
```diff
@@ -10,7 +10,7 @@
     "removeComments": true,
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "strictNullChecks": true
```
Comment (rhysbarrett) [inline tsconfig.json:13]: How come we're removing this flag?

File: src/controllers/StartPageController.ts (hunk 4 above)
Comment (akaJSaunders) [reply src/controllers/StartPageController.ts:10]: controls the behavior of the post request from the start page. So when you click the start button you will be redirected to ROOT_URI (currently just refreshes page). 

This was handy for me to test the session

File: tsconfig.json (hunk 5 above)
Comment (akaJSaunders) [reply tsconfig.json:13]: Not sure I really understand what that discussion is about, why did you pick it?

File: tsconfig.json (hunk 6 above)
Comment (akaJSaunders) [reply tsconfig.json:13]: I think it was by mistake :P

File: src/controllers/StartPageController.ts (hunk 4 above)
Comment (rhysbarrett) [reply src/controllers/StartPageController.ts:10]: Ah I see. Thanks for explaining.

File: tsconfig.json (hunk 5 above)
Comment (rhysbarrett) [reply tsconfig.json:14]: It was too long ago to remember precisely, but I'm pretty sure it was recommended by the linter when I tried using a certain type of import, so I looked it up. Judging by the discussion, the two parameters do more or less the same thing, so I'm not picky about which one gets used, but we don't need both.

File: src/app.ts (hunk 1 above)
Comment (akaJSaunders) [reply src/app.ts:8]: I have been having a lot of issues with getting this cookie parser, for some reason when I do it the way you suggested, the module is not indentified in runtime. Ill have to take a look at it

**Pull request closed without merging**